fenix/app/src/test/java/org/mozilla/fenix/telemetry/TelemetryMiddlewareTest.kt

```diff
@@ -60,12 +63,21 @@ class TelemetryMiddlewareTest {
 
     private val clock = FakeClock()
     private val metrics: MetricController = mockk()
+    private val searchState: MutableMap<String, TimerId> = mutableMapOf()
+    private val timerId = Metrics.searchPageLoadTime.start()
 
     @Before
     fun setUp() {
         Clock.delegate = clock
         settings = Settings(testContext)
-        telemetryMiddleware = TelemetryMiddleware(testContext, settings, metrics)
+        telemetryMiddleware = TelemetryMiddleware(
+            context = testContext,
+            settings = settings,
+            metrics = metrics,
+            nimbusSearchEngine = SEARCH_ENGINE_NAME,
+            searchState = searchState,
+            timerIdFunc = { timerId },
```

Comment:
We might not even need this as a function injected in the constructor as we're using the same value in the tests 🤔

Reply:
As far as I can see, we need to pass the TimerId from the tests to be able to assert the correct value, but looking at it again it doesn't need to be a fun reference 👍